### PR TITLE
Migrating u-boot envars from NOR to SD

### PR DESCRIPTION
**Pull request by CatFreed**
In an effort to reduce the number of NOR flash writes, the U-Boot environment variable storage area has been moved onto the SD card, into the user data partition. It will be available from the Linux environment as a new file `/home/system/etc/uboot.env`

Additionally, I added a new "user-overlay" folder to this repo and a step in the SD image creation script to apply this overlay to the user partition. I needed it to create the "/home/system/etc" folder, but have also added all the other currently expected user partition folders.

Made in conjunction with kubostech/uboot#12

### CODE REVIEW

**Comment by CatFreed** [comment]
[Migrate U-Boot envars to SD card](https://trello.com/c/wxszCkfx/36-migrate-u-boot-envars-to-sd-card)

File: tools/format-image.sh [hunk 1]
```diff
@@ -147,6 +147,12 @@ if [ "${package}" -gt "0" ]; then
   sleep 1
   umount /dev/loop0p6
 
+  echo '\nCreating the user partition overlay'
+  mount /dev/loop0p1 /tmp-kubos
+  cp ../user-overlay/* /tmp-kubos -R
+  sleep 1
```
Comment (kyleparrott) [inline tools/format-image.sh]: Is the `sleep` call necessary?

Comment (CatFreed) [inline tools/format-image.sh]: No idea. @plauche put them in the previous steps, so I just copied that format. Certainly doesn't hurt, though

Comment (kyleparrott) [inline tools/format-image.sh]: Cooooool 🍨

**Pull request merged**